File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        unordered_set<char>st;
        int n=s.size();
        int j=0,ans=0;
        for(int i=0;i<n;i++){
            while(st.count(s[i])){
                st.erase(s[j]);
                j++;
            }
            st.insert(s[i]);
            ans=max(ans,(int)st.size());
        }
        return ans;
    }
};

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution(object):
    def lengthOfLongestSubstring(self, s):
        st=set()
        n=len(s)
        j=0
        ans=0
        for i in range(n):
            while s[i] in st:
                st.remove(s[j])
                j+=1
            st.add(s[i])
            ans=max(ans,len(st))
        return ans